src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -6,295 +6,193 @@
-          graphics.drawImage(movementImage, (int) (points[numTerritories - 1].x + textXOffset - xOffset - translateX),
-              (int) (points[numTerritories - 1].y + textyOffset - yOffset - translateY), null);
-        }
-        final Image cursorImage = routeDescription.getCursorImage();
-        if (cursorImage != null) {
-          graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-              (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          if (scrollWrapX /* && !scrollWrapY */) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset), null);
-          }
-          if (/* !scrollWrapX && */scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset),
-                (int) (points[numTerritories - 1].y + cursorYOffset - yOffset - translateY), null);
-          }
-          if (scrollWrapX && scrollWrapY) {
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset + translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset + translateY), null);
-            graphics.drawImage(cursorImage, (int) (points[numTerritories - 1].x + cursorXOffset - xOffset - translateX),
-                (int) (points[numTerritories - 1].y + cursorXOffset - yOffset - translateY), null);
-          }
-        }
-      }
-    } finally {
-      graphics.setTransform(original);
+    if (routeDescription == null) {
+      return;
+    }
+    final Route route = routeDescription.getRoute();
+    if (route == null) {
+      return;
+    }
+
+    final Point[] points = getRoutePoints(routeDescription, mapData);
+    final int xOffset = view.getXOffset();
+    final int yOffset = view.getYOffset();
+    final double scale = view.getScale();
+    final int jointsize = 10;
+    final int numTerritories = route.getAllTerritories().size();
+    // set thickness and color of the future drawings
+    graphics.setStroke(new BasicStroke(3.5f, BasicStroke.CAP_ROUND, BasicStroke.JOIN_ROUND));
+    graphics.setPaint(Color.red);
+    graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+
+    if (Arrays.asList(points).contains(null)) {// If the Array is null at some point
```

Comment:
I think this is a legacy code snippet when I ran into errors with the points array...
Can be removed I guess...